src/components/explore/VenueReactButton.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+import { Text, TouchableOpacity } from 'react-native';
+import { ReactionEmoji, VenueReactButtonProps } from '@nightlight/src/types';
+import VenueReactButtonStyles from '@nightlight/components/explore/VenueReactButton.styles';
+
+const VenueReactButton = ({ venue, reaction }: VenueReactButtonProps) => {
+  const [count, setCount] = useState(venue.reactions[reaction].count);
+  const [didReact, setDidReact] = useState(venue.reactions[reaction].didReact);
+
+  const handleToggleReaction = (reaction: ReactionEmoji) => {
+    // Preemptively update the count to avoid a delay in the UI
+    setCount(prev => (didReact ? prev - 1 : prev + 1));
+    setDidReact(prev => !prev);
+
+    // TODO: Asynchronously update reaction in DB
+    console.log(
+      `TODO: Toggle reaction for ${reaction} on venue ${venue._id} for user _!`
+    );
+
+    // TODO: Revert the UI update if the DB update fails
```

Comment:
If DB update fails we should also show the error message similar to what we have in signup/login page. 
(future TODO though)